diff --git a/pam_authz/pam/flag.c b/pam_authz/pam/flag.c
index 447faf6..5fb6fe8 100644
--- a/pam_authz/pam/flag.c
+++ b/pam_authz/pam/flag.c
@@ -4,10 +4,11 @@
 #include "flag.h"
 #include "log.h"
 
-#define FLAG_COUNT 5
+#define FLAG_COUNT 6
 
 // Valid flags that can be used as arguments to this PAM module inside /etc/pam.d/ files.
 static const char FLAG_STR_OPA_URL[]          = "url";
+static const char FLAG_STR_OPA_SOCK[]         = "sock";
 static const char FLAG_STR_PULL_ENDPOINT[]    = "pull_endpoint";
 static const char FLAG_STR_DISPLAY_ENDPOINT[] = "display_endpoint";
 static const char FLAG_STR_AUTHZ_ENDPOINT[]   = "authz_endpoint";
@@ -15,6 +16,7 @@ static const char FLAG_STR_LOG_LEVEL[]        = "log_level";
 
 // Default values of exposed variables.
 char flag_opa_url[MAX_FLAG_SIZE]          = "";
+char flag_opa_sock[MAX_FLAG_SIZE]         = "";
 char flag_pull_endpoint[MAX_FLAG_SIZE]    = ""; 
 char flag_display_endpoint[MAX_FLAG_SIZE] = "";
 char flag_authz_endpoint[MAX_FLAG_SIZE]   = "";
@@ -25,6 +27,7 @@ static const struct FlagStrToVar {
 	const char *var;
 } FLAG_STR_TO_VAR[FLAG_COUNT] = {
 	{FLAG_STR_OPA_URL,          flag_opa_url},
+	{FLAG_STR_OPA_SOCK,         flag_opa_sock},
 	{FLAG_STR_PULL_ENDPOINT,    flag_pull_endpoint},
 	{FLAG_STR_DISPLAY_ENDPOINT, flag_display_endpoint},
 	{FLAG_STR_AUTHZ_ENDPOINT,   flag_authz_endpoint},
@@ -79,4 +82,4 @@ void initialize_flags(int argc, const char **argv) {
 		strcpy((char *)flag_var, value);
 		free(tofree);
 	}
-}
\ No newline at end of file
+}
diff --git a/pam_authz/pam/flag.h b/pam_authz/pam/flag.h
index 9f7fc5c..baeab7e 100644
--- a/pam_authz/pam/flag.h
+++ b/pam_authz/pam/flag.h
@@ -6,6 +6,7 @@
 // These variables are populated by initialize_flags() using arguments
 // that this PAM module is called with.
 extern char flag_opa_url[MAX_FLAG_SIZE];
+extern char flag_opa_sock[MAX_FLAG_SIZE];
 extern char flag_pull_endpoint[MAX_FLAG_SIZE];
 extern char flag_display_endpoint[MAX_FLAG_SIZE];
 extern char flag_authz_endpoint[MAX_FLAG_SIZE];
@@ -16,4 +17,4 @@ extern char flag_log_level[MAX_FLAG_SIZE];
 extern void
 initialize_flags(int argc, const char **argv);
 
-#endif
\ No newline at end of file
+#endif
diff --git a/pam_authz/pam/http.c b/pam_authz/pam/http.c
index e82907a..209fa43 100644
--- a/pam_authz/pam/http.c
+++ b/pam_authz/pam/http.c
@@ -67,6 +67,9 @@ int http_request(const char * method, const char* endpoint, char *req_body, json
 	}
 
 	char url[2 * MAX_FLAG_SIZE];
+	if (strlen(flag_opa_sock) > 0) {
+		curl_easy_setopt(curl_handle, CURLOPT_UNIX_SOCKET_PATH, flag_opa_sock);
+	}
 	curl_easy_setopt(curl_handle, CURLOPT_URL, join_url(url, flag_opa_url, endpoint));
 	curl_easy_setopt(curl_handle, CURLOPT_CUSTOMREQUEST, method);
 	curl_easy_setopt(curl_handle, CURLOPT_NOPROGRESS, 1);
@@ -152,4 +155,4 @@ int http_request(const char * method, const char* endpoint, char *req_body, json
 	}
 
 	return resp_code;
-}
\ No newline at end of file
+}
